realm/realm-library/src/main/cpp/io_realm_SyncManager.cpp

```diff
@@ -44,7 +50,23 @@ struct AndroidClientListener : public realm::BindingCallbackThreadObserver {
         // Failing to detach the JVM before closing the thread will crash on ART
         JniUtils::detach_current_thread();
     }
-} s_client_thread_listener;
+
+    void handle_error(std::exception const& e) override
+    {
+        JNIEnv* env = JniUtils::get_env(true);
+        std::string msg = format("An exception has been thrown on the sync client thread:\n%1", e.what());
+        Log::f(msg.c_str());
+        // Since user has no way to handle exceptions thrown on the sync client thread, we just convert it to an Java
+        // exception to get more debug information for ourself.
+        // FIXME: We really need to find a universal and clever way to get the native backtrace when exception thrown
+        env->ThrowNew(m_realm_exception_class, msg.c_str());
+    }
+
+private:
+    // For some reasons, FindClass() doesn't work in the native thread even when the JVM is attached before. Get the
```

Comment:
Is it a bug in Android? Is it a know bug? Or does it also apply to any JVM implementation?